CMakeLists.txt

```diff
@@ -150,6 +160,23 @@ if (${TRITON_PYTORCH_DOCKER_BUILD})
 
   string(REPLACE ";" " " CONDA_LIBS_STR "${CONDA_LIBS}")
 
+  if (${TRITON_PYTORCH_ENABLE_TORCHTRT})
+    execute_process(
```

Comment:
@borisfom you should remove the auto detecting of TorchTRT and changing the cmake flag. We should **not silently change** the setting to make it valid. The pytorch backend should follow the pattern that other backends have and be consistent with how they handle that. Eg: https://github.com/triton-inference-server/onnxruntime_backend/blob/main/CMakeLists.txt
ORT backend as well has a flag for enabling TRT support: TRITON_ENABLE_ONNXRUNTIME_TENSORRT and has the checks I have asked you to add. It can be ON by default. But if that setting conflicts with other settings, like TRITON_ENABLE_GPU, then we should raise an error. 

Please make these changes so that we adhere to the existing patterns/policies Triton has.

Reply:
"It can be ON by default. But if that setting conflicts with other settings, like TRITON_ENABLE_GPU, then we should raise an error." - that's a usability problem. It has to be OFF by default if you want to enforce it.

Reply:
I don't think TRITON_PYTORCH_ENABLE_TORCHTRT has to be consistent with TRITON_ENABLE_ONNXRUNTIME_TENSORRT - those are very different settings. For ONNXruntime - it changes basic execution, and therefore is OFF by default. TRITON_PYTORCH_ENABLE_TORCHTRT is only extending support, not changing anything. So we think it has to be ON by default. This does not even have to be an option - the only reason user may want to turn it OFF when it's available is to save 500K of memory if he is sure he is not going to need it.

Reply:
We can treat them the same though - if you think we should be consistent here, I will flip the default to OFF and we'll set the flag to ON in server's build.py. I would leave autodetect-off-switch on and issue a warning to handle the case someone would be building with latest code using PyTorch image from a few months back.

Reply:
I understand but based on the Triton policies/patterns we should be not setting these options automatically and that is a design choice the Triton team has taken and wishes to stick to.